### PR TITLE
[MO] Sqrt operation implementation

### PR DESCRIPTION
**Pull request by achetver**
Root cause analysis: MO haven't shape and value propagation function for `Sqrt` operation. For restored by MO IR Reader graph default shape propagation function is used and it leads to error while we try to propagate value to calculate for second input for `Reshape` operation.

Solution: Add implementation of `Sqrt` operation. 

Ticket: 76784

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names
* [x]  Transformation preserves tensor names

Validation:
* [x]  Unit tests: N/A - Shape inference is so simple and we don't implement tests for such functions;
* [x]  Framework operation tests: N/A - No new enabled operations;
* [x]  Transformation tests: N/A - No new implemented transformations;
* [x]  Model Optimizer IR Reader check: Done manually.

Documentation:
* [x]  Supported frameworks operations list: N/A - No new supported operations;
* [x]  Supported **public** models list: N/A - Nothing to add;
* [x]  User guide update: N/A - Nothing to update.

### CODE REVIEW

File: tools/mo/openvino/tools/mo/utils/ir_reader/extenders/sqrt_extender.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+from openvino.tools.mo.front.common.partial_infer.utils import float32_array
+from openvino.tools.mo.utils.graph import Node
+from openvino.tools.mo.utils.ir_reader.extender import Extender
+
+
+class Sqrt_extender(Extender):
+    op = 'Sqrt'
+
+    @staticmethod
+    def operation(a):   # we use same function as for elementwise.power but with fixed
```
Comment (achetver) [inline tools/mo/openvino/tools/mo/utils/ir_reader/extenders/sqrt_extender.py:16]: Update comment

Comment (achetver) [reply tools/mo/openvino/tools/mo/utils/ir_reader/extenders/sqrt_extender.py]: Done

File: tools/mo/openvino/tools/mo/utils/ir_reader/extenders/sqrt_extender.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+from openvino.tools.mo.front.common.partial_infer.utils import float32_array
+from openvino.tools.mo.utils.graph import Node
+from openvino.tools.mo.utils.ir_reader.extender import Extender
+
+
+class Sqrt_extender(Extender):
+    op = 'Sqrt'
+
+    @staticmethod
+    def operation(a):   # we use same function as for elementwise.power but with fixed зщцук = 0.5
```
Comment (vgavrilo) [inline tools/mo/openvino/tools/mo/utils/ir_reader/extenders/sqrt_extender.py:16]: Maybe, `power = 0.5`?

**Review by vgavrilo — CHANGES_REQUESTED** [review]

File: tools/mo/openvino/tools/mo/utils/ir_reader/extenders/sqrt_extender.py (hunk 2 above)
Comment (achetver) [reply tools/mo/openvino/tools/mo/utils/ir_reader/extenders/sqrt_extender.py]: Fixed.

**Comment by achetver** [comment]
@mvafin @sadolini Please, review this PR.

**Comment by achetver** [comment]
@mvafin @sadolini Please, review this PR.

**Review by mvafin — APPROVED** [review]

**Review by sadolini — APPROVED** [review]

**Comment by achetver** [comment]
@rkazants Please, take a look at this PR.

File: tools/mo/openvino/tools/mo/utils/ir_reader/extenders/sqrt_extender.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+    @staticmethod
+    def operation(a):   # we use same function as for elementwise.power but with fixed power = 0.5
+        if np.issubdtype(a.dtype, np.signedinteger):
+            return float32_array(a.astype(np.float32) ** 0.5)
+        return a ** 0.5
+
+    @staticmethod
+    def extend(op: Node):
+        op['infer'] = lambda node: eltwise_infer(node, Sqrt_extender.operation)
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/utils/ir_reader/extenders/sqrt_extender.py:23]: let us add `Sqrt` shape and value infer propagation in `ops` directory please.
It is useful routine and can become useful in future.

Comment (achetver) [reply tools/mo/openvino/tools/mo/utils/ir_reader/extenders/sqrt_extender.py]: Done.

File: tools/mo/openvino/tools/mo/ops/elementwise.py [hunk 4]
```diff
@@ -234,3 +234,14 @@ class Negative(UnaryElementwise):
...
+
+class Sqrt(UnaryElementwise):
+    op = 'Sqrt'
+    op_type = 'Sqrt'
+
+    @staticmethod
+    def operation(a):
+        if np.issubdtype(a.dtype, np.signedinteger):
+            return float32_array(a.astype(np.float32) ** 0.5)
+        return a ** 0.5
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/ops/elementwise.py:247]: in both cases the result output type must be the same as input type. Please check

[the remark this answers is not in the document]
Comment (achetver) [reply tools/mo/openvino/tools/mo/ops/elementwise.py:247]: The same value calculation function as for `Power` operation was used.
I think the only problem can be with `int` input - output will be with `float` type. But now we have the same behavior for `Power`. So I think we can use it or we need to change it together with `Power` operation.

[the remark this answers is not in the document]
Comment (rkazants) [reply tools/mo/openvino/tools/mo/ops/elementwise.py:247]: I see. Thanks for the explanation. In that model Sqrt operates on FP64 type so it looks no impact.

**Review by rkazants — APPROVED** [review]

**Pull request merged**